solver/pb/ops.proto

```diff
@@ -111,7 +112,14 @@ message OpMetadata {
 
 message ExportCache {
 	bool Value = 1;
-};
+}
+
+message ProxyEnv {
```

Comment:
Can we make this less opinionated, e.g. `repeated string uncached_env`?

Reply:
Why do you prefer that? I think its a problem with the current implementation in moby that these are treated as "special build arguments" while really they are just some worker configuration that should probably be applied per worker instead. Also, if we add support for these for image, http and git source (left out atm as Moby doesn't do that) then these values are not env variables at all.

Reply:
My first though was that potentially 3rd party languages might want to exclude other env vars from the cache key as well , but +1 for `http and git source (left out atm as Moby doesn't do that) then these values are not env variables at all`.

So LGTM when CI gets green.

Reply:
Restarted. It is the same issue that is being debugged in #399